tfe/resource_tfe_agent_pool.go

```diff
@@ -0,0 +1,107 @@
+package tfe
+
+import (
+	"fmt"
+	"log"
+
+	tfe "github.com/hashicorp/go-tfe"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+)
+
+func resourceTFEAgentPool() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceTFEAgentPoolCreate,
+		Read:   resourceTFEAgentPoolRead,
+		Update: resourceTFEAgentPoolUpdate,
+		Delete: resourceTFEAgentPoolDelete,
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+
+			"organization": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+		},
+	}
+}
+
+func resourceTFEAgentPoolCreate(d *schema.ResourceData, meta interface{}) error {
+	tfeClient := meta.(*tfe.Client)
+
+	// Get the name and organization.
+	name := d.Get("name").(string)
+	organization := d.Get("organization").(string)
+
+	// Create a new options struct.
+	options := tfe.AgentPoolCreateOptions{
+		Name: tfe.String(name),
+	}
+
+	log.Printf("[DEBUG] Create new agent pool for organization: %s", organization)
+	agentPool, err := tfeClient.AgentPools.Create(ctx, organization, options)
+	if err != nil {
+		return fmt.Errorf(
+			"Error creating agent pool %s for organization %s: %v", name, organization, err)
+	}
+
+	d.SetId(agentPool.ID)
+
+	return resourceTFEAgentPoolUpdate(d, meta)
```

Comment:
i think this should be a call to `resourceTFEAgentPoolRead`
```suggestion
	return resourceTFEAgentPoolRead(d, meta)
```

Reply:
It "should be" update, based on the [template I originally borrowed from](https://github.com/hashicorp/terraform-provider-tfe/blob/0e8c31650fd1ef963226b96e6b2c03d0e5f4098d/tfe/resource_tfe_ssh_key.go#L61)

I'm guessing this is another case of intended before I really don't like, [which I discovered in another PR recently](https://github.com/hashicorp/terraform-provider-tfe/pull/239#discussion_r525373874). 😕

This is all to say I agree, and think we should fix those other places when time permits 😑  Good catch.